src/components/sidebar/sidebar.component.tsx

```diff
@@ -1,134 +1,91 @@
 
 interface SidebarProps {
-  allowUnspecifiedAll: boolean;
   defaultPage: string;
-  handleClose: () => void;
-  hideFormCollapseToggle: () => void;
   isFormSubmitting: boolean;
-  mode: string;
+  sessionMode: SessionMode;
   onCancel: () => void;
-  pagesWithErrors: string[];
-  scrollablePages: Set<FormPage>;
-  selectedPage: string;
-  setValues: (values: unknown) => void;
-  values: object;
+  handleClose: () => void;
+  hideFormCollapseToggle: () => void;
 }
 
 const Sidebar: React.FC<SidebarProps> = ({
-  allowUnspecifiedAll,
   defaultPage,
-  handleClose,
-  hideFormCollapseToggle,
   isFormSubmitting,
-  mode,
+  sessionMode,
   onCancel,
-  pagesWithErrors,
-  scrollablePages,
-  selectedPage,
-  setValues,
-  values,
+  handleClose,
+  hideFormCollapseToggle,
 }) => {
   const { t } = useTranslation();
-  const pages: Array<FormPage> = Array.from(scrollablePages);
-
-  useEffect(() => {
-    if (defaultPage && pages.some(({ label, isHidden }) => label === defaultPage && !isHidden)) {
-      scrollIntoView(joinWord(defaultPage));
-    }
-  }, [defaultPage, scrollablePages]);
-
-  const unspecifiedFields = useMemo(
-    () =>
-      Object.keys(values).filter(
-        (key) => key.endsWith('-unspecified') && isEmpty(values[key.split('-unspecified')[0]]),
-      ),
-    [values],
-  );
-
-  const handleClick = (selected) => {
-    const activeId = joinWord(selected);
-    scrollIntoView(activeId);
-  };
-
-  const markAllAsUnspecified = useCallback(
-    (toggled) => {
-      const updatedValues = { ...values };
-      unspecifiedFields.forEach((field) => {
-        updatedValues[field] = toggled;
-      });
-      setValues(updatedValues);
-    },
-    [unspecifiedFields, values, setValues],
-  );
+  const { pages, pagesWithErrors, activePages, evaluatedPagesVisibility } = usePageObserver();
+  const { currentActivePage, requestPage } = useCurrentActivePage({
+    pages,
+    defaultPage,
+    activePages,
+    evaluatedPagesVisibility,
+  });
 
   return (
     <div className={styles.sidebar}>
-      {pages.map((page, index) => {
-        if (page.isHidden) return null;
+      {pages
+        .filter((page) => isPageContentVisible(page))
+        .map((page) => {
+          const isActive = page.id === currentActivePage;
+          const hasError = pagesWithErrors.includes(page.id);
+          return (
+            <div
+              className={classNames(styles.tab, {
+                [styles.activeTab]: isActive && !hasError,
+                [styles.errorTab]: hasError && !isActive,
+                [styles.activeErrorTab]: hasError && isActive,
+              })}>
+              <button
+                onClick={(e) => {
+                  e.preventDefault();
+                  requestPage(page.id);
+                }}>
+                <span>{page.label}</span>
+              </button>
+            </div>
+          );
+        })}
+      {sessionMode !== 'view' && <hr className={styles.divider} />}
```

Comment:
There might be a use case for `embedded-view` mode within the active visits that might necessitate this being updated.

Reply:
I'm not sure of the use case but currently, the sidenav isn't supported in "embedded-view"